easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,362 @@
+        self.cfg['srcdir'] = os.path.join(self.start_dir, 'cmake')
+
+        # verbose CMake
+        if '-DCMAKE_VERBOSE_MAKEFILE=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DCMAKE_VERBOSE_MAKEFILE=yes')
+
+        # Enable following packages, if not configured in easycofig
+        default_options = [
+            'BUILD_DOC', 'BUILD_EXE', 'BUILD_LIB',
+            'BUILD_SHARED_LIBS', 'BUILD_TOOLS',
+        ]
+        for option in default_options:
+            if "-D%s=" % option not in self.cfg['configopts']:
+                self.cfg.update('configopts', '-D%s=on' % option)
+
+        # Is there a gzip
+        if '-DWITH_GZIP=' not in self.cfg['configopts']:
+            if get_software_root('gzip'):
+                self.cfg.update('configopts', '-DWITH_GZIP=yes')
+            else:
+                self.cfg.update('configopts', '-DWITH_GZIP=no')
+
+        # Is there a libpng
+        if '-DWITH_PNG=' not in self.cfg['configopts']:
+            if get_software_root('libpng'):
+                self.cfg.update('configopts', '-DWITH_PNG=yes')
+            else:
+                self.cfg.update('configopts', '-DWITH_PNG=no')
+
+        # Is there a libjpeg-turbo
+        if '-DWITH_JPEG=' not in self.cfg['configopts']:
+            if get_software_root('libjpeg-turbo'):
+                self.cfg.update('configopts', '-DWITH_JPEG=yes')
+            else:
+                self.cfg.update('configopts', '-DWITH_JPEG=no')
+
+        # With Eigen dependency:
+        if '-DDOWNLOAD_EIGEN3=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DDOWNLOAD_EIGEN3=no')
+        # Compiler complains about 'Eigen3_DIR' not beeing set, but acutally it needs
+        # 'EIGEN3_INCLUDE_DIR'.
+        # see: https://github.com/lammps/lammps/issues/1110
+        if '-DEIGEN3_INCLUDE_DIR=' not in self.cfg['configopts']:
+            if get_software_root('Eigen'):
+                self.cfg.update('configopts', '-DEIGEN3_INCLUDE_DIR=%s/include/Eigen' % get_software_root('Eigen'))
+
+        if '-DEigen3_DIR=' not in self.cfg['configopts']:
+            if get_software_root('Eigen'):
+                self.cfg.update('configopts', '-DEigen3_DIR=%s/share/eigen3/cmake/' % get_software_root('Eigen'))
+
+        # LAMMPS Configuration Options
+        # https://github.com/lammps/lammps/blob/master/cmake/README.md#lammps-configuration-options
+        if self.cfg['general_packages']:
+            for package in self.cfg['general_packages']:
+                self.cfg.update('configopts', '-DPKG_%s=on' % package)
+
+        if self.cfg['user_packages']:
+            for package in self.cfg['user_packages']:
+                self.cfg.update('configopts', '-DPKG_USER-%s=on' % package)
+
+        # Optimization settings
+        if '-DPKG_OPT=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DPKG_OPT=on')
+
+        if '-DPKG_USR-INTEL=' not in self.cfg['configopts']:
```

Comment:
You would need to be using an Intel compiler for this to be valid so best to check the toolchain compiler family before setting this.

Reply:
I might get that wrong, but from what I've read I think that you don't need intel compilers for this: https://lammps.sandia.gov/doc/Speed_intel.html 
Those should be able to run being compiled with GCC as well.